File: tasks/3541-find-most-frequent-vowel-and-consonant.ts
// Accepted 812 / 812 testcases passed Sergey Pomortsev submitted at Feb 01, 2026 14:48
// Solution Runtime 2 ms Beats 95.90% Analyze Complexity Memory 57.77 MB Beats 83.61%

const vovels = ['a', 'e', 'i', 'o', 'u']

function maxFreqSum(s: string): number {
  const mapV = new Map()
  let maxVCount = 0
  let maxV
  const mapC = new Map()
  let maxCCount = 0
  let maxC

  s.split('').forEach(el => {
    const isV = vovels.includes(el)
    const map = isV ? mapV : mapC
    const maxCount = isV ? maxVCount : maxCCount

    if (map.has(el)) map.set(el, map.get(el) + 1)
    else map.set(el, 1)

    if (map.get(el) > maxCount) {
      if (isV) maxVCount = map.get(el)
      else maxCCount = map.get(el)

      if (isV) maxV = el
      else maxC = el
    }
  })

  return maxVCount + maxCCount
};

File: tasks/3541-find-most-frequent-vowel-and-consonant.go
// Accepted 812 / 812 testcases passed Sergey Pomortsev submitted at Jul 12, 2026 10:33
// Runtime 0 ms Beats 100.00% Memory 4.41 MB Beats 13.33%

func maxFreqSum(s string) int {
  var vowels = map[string]struct{}{"a": {}, "e": {}, "i": {}, "o": {}, "u": {}}
  var collection = make(map[string]int)
  var maxFreqVowels, maxFreqConsonant int

  for _, symbRune := range s {
    var symb = string(symbRune)
    if _, ok := collection[symb]; ok {
      collection[symb] += 1
    } else {
      collection[symb] = 1
    }
  }

  for key, freq := range collection {
    var _, isVowel = vowels[key]
    if isVowel && maxFreqVowels < freq {
      maxFreqVowels = freq
    } else if !isVowel && maxFreqConsonant < freq {
      maxFreqConsonant = freq
    }
  }

  return maxFreqVowels + maxFreqConsonant
}